src/Docker/Manager/ContainerManager.php

```diff
@@ -210,6 +210,72 @@ public function run(Container $container, callable $attachCallback = null, array
     }
 
     /**
+     * Execute a command in a running container
+     * i.e. create an executive, which will be run by execstart()
+     *
+     * @param \Docker\Container     $container
+     * @param array    $cmd          command to run
+     * @param boolean  $attachstdin  
+     * @param boolean  $attachstdout 
+     * @param boolean  $attachstderr 
+     * @param integer  $tty  
+     *
+     * @throws \Docker\Exception\UnexpectedStatusCodeException
+     *
+     * @return ID of the executive
+     */
+    public function exec(Container $container, array $cmd = [], $attachstdin = false, $attachstdout = true, $attachstderr = true, $tty = false)
+    {
+        $body = [
+            'AttachStdin'  => $attachstdin,
+            'AttachStdout' => $attachstdout,
+            'AttachStderr' => $attachstderr,
+            'Tty'          => $tty,
+            'Cmd' => $cmd
+        ];
+        $response = $this->client->post(['/containers/{id}/exec', ['id' => $container->getId()]], [
+            'body'         => Json::encode($body),
+            'headers'      => ['content-type' => 'application/json'],
+        ]);
+
+        if ($response->getStatusCode() !== "201") {
+            throw UnexpectedStatusCodeException::fromResponse($response);
+        }
+
+      return $response->json()['Id'];
+    }
+
+    /**
+     * Start an executive defined from exec()
+     * This can be resude several times, so if the command /bin/date in defined in exec()
+     * execstart() on that ID will return a different value each time.
+     * todo: how are instances created by exec() and used by execstart() removed/cleanedup?
+     *
+     * @param string   $id       identifier from exec()
```

Comment:
Should be `$execid`

Maybe rename that to $execId to be in line with the code (for example `$containerId`) !?